src/Symfony/Component/DependencyInjection/Compiler/AutoAliasServicePass.php

```diff
@@ -33,7 +33,7 @@ public function process(ContainerBuilder $container)
 
                 $aliasId = $container->getParameterBag()->resolveValue($tag['format']);
                 if ($container->hasDefinition($aliasId) || $container->hasAlias($aliasId)) {
-                    $container->setAlias($serviceId, new Alias($aliasId, true));
+                    $container->setAlias($serviceId, new Alias($aliasId, $container->getDefinition($serviceId)->isPublic()));
```

Comment:
this should come with a BC layer that triggers a deprecation when the alias is used as a public one while the target is private

Reply:
@nicolas-grekas Done, if `public` is not explicitly set it will default to `true` and trigger a deprecation. In the future the deprecation can be removed and `false` set as the default.

Reply:
I was thinking about something like https://symfony.com/blog/new-in-symfony-5-1-deprecate-public-services-into-private-services but for aliases. Not sure how. /cc @fancyweb